ts/features/zendesk/components/ZendeskSupportComponent.tsx

```diff
@@ -91,6 +93,9 @@ const ZendeskSupportComponent = (props: Props) => {
       .getOrElse({});
 
     setUserIdentity(zendeskIdentity);
+    hasOpenedTickets().then((n: number) =>
+      setShowAlreadyOpenedTicketButton(n > 0)
+    );
```

Comment:
```suggestion
    hasOpenedTickets().then((n: number) =>
      setShowAlreadyOpenedTicketButton(n > 0)
    );
```
This code could update the state when the component is unmounted, furthermore it could cause memory leaks
This is a possible scenario (time ordered):
- ask for tickets number
- close the component
- have response about tickets
- try to update a state of unmounted component

This is not the best but it's a fast solution (put this logic in a dedicated effect)
```typescript
useEffect(() => {
    // eslint-disable-next-line functional/no-let
    let isMounted = true;
    hasOpenedTickets().then((n: number) => {
      if (isMounted) {
        setShowAlreadyOpenedTicketButton(n > 0);
      }
    });
    return () => {
      isMounted = false;
    }; // cleanup toggles value, if unmounted
  }, []);
```

Reply:
Thank you! I didn't notice this issue!
I refactored using redux-saga and adding the associated store section.
These are the commits: [7c46353](https://github.com/pagopa/io-app/pull/3605/commits/7c46353ea2a2c10b8e6a31377ec852e35d68c280), [d5e1a22](https://github.com/pagopa/io-app/pull/3605/commits/d5e1a22e44bd9f4f208d334d4d24ae65a5d0feed), [1fe5758](https://github.com/pagopa/io-app/pull/3605/commits/1fe575800e274bc2e4b733d75e1aa8ad3ff68b01), [d128eb8](https://github.com/pagopa/io-app/pull/3605/commits/d128eb82d70c01a158c41008e17db3f1cdfb35d8)

Reply:
Your solution is even better!